akka-docs/rst/java/futures.rst

```diff
@@ -260,3 +260,87 @@ After
 
 .. includecode:: code/docs/future/FutureDocTest.java
    :include: after
+
+Java 8, CompletionStage and CompletableFuture
+---------------------------------------------
+
+Starting with Akka 2.4.2 we have begun to introduce Java 8 ``java.util.concurrent.CompletionStage`` in Java APIs.
+It's a ``Future`` counterpart in Java; conversion from ``Future`` is done using ``scala-java8-compat`` library.
+
+Unlike ``Future``, ``CompletionStage`` has *async* and *non-async* methods.
+
+``scala-java8-compat`` library returns own implementation of ``CompletionStage`` which delegates all *non-async*
+methods to their *async* counterparts. The implementation extends standard Java ``CompletableFuture``.
+Java 8 ``CompletableFuture`` creates a new instance of ``CompletableFuture`` for any new stage,
+which means ``scala-java8-compat`` implementation is not used after the first mapping method.
+
+.. note::
```

Comment:
Reword the "so, by Akka" sounds a bit colloquial. 
Perhaps "e.g. CompletionStage returned by Akka" would be good.